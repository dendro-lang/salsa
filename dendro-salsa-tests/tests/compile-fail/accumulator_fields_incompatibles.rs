#[salsa::jar(db = Db)]
struct Jar(AccTwoUnnamedFields, AccNamedField);

trait Db: salsa::DbWithJar<Jar> {}

// accumulator with more than one unnamed fields
#[salsa::accumulator(jar = Jar)]
struct AccTwoUnnamedFields (u32, u32);


// accumulator with named fields
#[salsa::accumulator(jar = Jar)]
struct AccNamedField {
    field: u32,
}

fn main() {}