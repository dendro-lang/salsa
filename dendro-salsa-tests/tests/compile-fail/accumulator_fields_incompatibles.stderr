error: accumulator structs should have only one anonymous field
 --> tests/compile-fail/accumulator_fields_incompatibles.rs:8:8
  |
8 | struct AccTwoUnnamedFields (u32, u32);
  |        ^^^^^^^^^^^^^^^^^^^

error: accumulator structs should have only one anonymous field
  --> tests/compile-fail/accumulator_fields_incompatibles.rs:13:8
   |
13 | struct AccNamedField {
   |        ^^^^^^^^^^^^^

error[E0412]: cannot find type `AccTwoUnnamedFields` in this scope
 --> tests/compile-fail/accumulator_fields_incompatibles.rs:2:12
  |
2 | struct Jar(AccTwoUnnamedFields, AccNamedField);
  |            ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `AccNamedField` in this scope
 --> tests/compile-fail/accumulator_fields_incompatibles.rs:2:33
  |
2 | struct Jar(AccTwoUnnamedFields, AccNamedField);
  |                                 ^^^^^^^^^^^^^ not found in this scope
